db/mongodb.go

```diff
@@ -0,0 +1,245 @@
+	db     *mongo.Database
+	ctx    context.Context
+}
+
+// NewMongoDB initializes a new MongoDB connection wrapped in a structure.
+func NewMongoDB(uri string) (MongoDB, error) {
+	opts := options.Client().ApplyURI(uri)
+	ctx := context.Background()
+	c, err := mongo.Connect(ctx, opts)
+	if err != nil {
+		return MongoDB{}, fmt.Errorf("failed to connect to MongoDB: %w", err)
+	}
+	if err := c.Ping(ctx, nil); err != nil {
+		return MongoDB{}, fmt.Errorf("failed to ping to MongoDB: %w", err)
+	}
+	u := strings.Split(uri, "?")[0] // Remove query parameters from the URI
+	ps := strings.Split(u, "/")
+	n := ps[len(ps)-1]
+	if n == "" || strings.Contains(n, "@") { // ensure the database name is valid
+		return MongoDB{}, fmt.Errorf("no database name found in the uri")
+	}
+	return MongoDB{
+		client: c,
+		db:     c.Database(n),
+		ctx:    ctx,
+	}, nil
+}
+
+// CreateCollection creates the specified collection if it does not already exist.
+func (m *MongoDB) CreateCollection() error {
+
+	if err := m.db.CreateCollection(m.ctx, companyTableName); err != nil {
+		return fmt.Errorf("error creating collection: %w", err)
+	}
+
+	log.Output(1, fmt.Sprintf("Collection %s created successfully", companyTableName))
+
+	if err := m.db.CreateCollection(m.ctx, metaTableName); err != nil {
+		return fmt.Errorf("erro ao criar a coleção: %w", err)
+	}
+
+	log.Output(1, fmt.Sprintf("Collection %s created successfully", metaTableName))
+
+	return nil
+}
+
+// CreateIndexes creates the indexes on the specified collection.
+func (m *MongoDB) CreateIndexes() error {
+	log.Output(1, "Creating the indexes...")
+	c := m.db.Collection(companyTableName)
+	i := []mongo.IndexModel{
+		{Keys: bson.D{{Key: "cnpj", Value: 1}}},
+	}
+	_, err := c.Indexes().CreateMany(m.ctx, i)
+	if err != nil {
+		return fmt.Errorf("error creating indexes: %w", err)
+	}
+	log.Output(1, fmt.Sprintf("Indexes successfully created in the collection: %s", companyTableName))
+	return nil
+}
+
+// DropCollection completely deletes a specific collection.
+func (m *MongoDB) DropCollection() error {
+	cs := []string{companyTableName, metaTableName}
+	for _, v := range cs {
+		c := m.db.Collection(v)
+
+		if err := c.Drop(m.ctx); err != nil {
+			return fmt.Errorf("error deleting collection: %w", err)
+		}
+
+		log.Output(1, fmt.Sprintf("Collection deleted successfully: %s", v))
+	}
+	m.Close()
+	return nil
+}
+
+// CreateCompanies writes a batch of company data to MongoDB
+func (m *MongoDB) CreateCompanies(batch [][]string) error {
+	if m == nil {
+		return fmt.Errorf("mongodb connection not initialized")
+	}
+	coll := m.db.Collection(companyTableName)
+	var cs []interface{} // required by MongoDb pkg
+	for _, r := range batch {
+		if len(r) < 2 {
+			return fmt.Errorf("line skipped due to insufficient length: %s", r)
+		}
+		var c company
+		c.Cnpj = r[0]
+		err := json.Unmarshal([]byte(r[1]), &c.Json)
+		if err != nil {
+			return fmt.Errorf("error deserializing JSON: %s, erro: %v", r[1], err)
```

Comment:
A forma correta de “envelopar” erros em Go é com `%w`:

```suggestion
			return fmt.Errorf("error deserializing JSON: %s, erro: %w", r[1], err)
```